src/main/java/io/fusionauth/samlv2/service/DefaultSAMLv2Service.java

```diff
@@ -313,7 +314,8 @@ public String buildAuthnResponse(AuthenticationResponse response, boolean sign,
 
       // AuthnStatement (element)
       AuthnStatementType authnStatement = new AuthnStatementType();
-      authnStatement.setAuthnInstant(toXMLGregorianCalendar(now));
+      ZonedDateTime authnInstant = response.authnInstant != null ? response.authnInstant : issueInstant;
```

Comment:
I'm not sure why we would default `authnInstant` to `issueInstant`. Shouldn't it default to `now`?
```suggestion
      ZonedDateTime authnInstant = response.authnInstant != null ? response.authnInstant : now;
```

Reply:
That is what I initially had, but I realized we were also not enforcing that the top level `issueInstant` was non-null. 

So I updated the top level `setIssueInstant` to be the value provided by the caller if non-null, else `now`. 

I figured if the caller is telling me when the response was issued, and they did not tell me when authentication occurred, I can assume it was at the same time as the statement was issued. That seems like a better assumption to me. 

In practice, we will always set `AuthnInstant` in our own usage.